agent.go

```diff
@@ -192,7 +192,6 @@ func (c *controller) agentSetup() error {
 	if remoteAddr != "" {
 		if err := c.agentJoin(remoteAddr); err != nil {
 			logrus.Errorf("Error in agentJoin : %v", err)
-			return nil
```

Comment:
Got it. Just noticed that even though the networkDB retry logic takes effect, call to `nDB.sendNodeEvent` is not involved in that retry. am not sure if that is required. Just giving my observation.